plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -347,3 +306,76 @@ export function decodePageCursor(pageCursor?: string): { page: number } {
 export function encodePageCursor({ page }: { page: number }): string {
   return Buffer.from(`${page}`, 'utf-8').toString('base64');
 }
+
+async function createElasticSearchClientOptions(
+  config?: Config,
+): Promise<ElasticSearchClientOptions> {
+  if (!config) {
+    throw new Error('No elastic search config found');
+  }
+  const clientOptionsConfig = config.getOptionalConfig('clientOptions');
+  const sslConfig = clientOptionsConfig?.getOptionalConfig('ssl');
+
+  if (config.getOptionalString('provider') === 'elastic') {
+    const authConfig = config.getConfig('auth');
+    return {
+      provider: 'elastic',
+      cloud: {
+        id: config.getString('cloudId'),
+      },
+      auth: {
+        username: authConfig.getString('username'),
+        password: authConfig.getString('password'),
+      },
+      ...(sslConfig
+        ? {
+            ssl: {
+              rejectUnauthorized:
+                sslConfig?.getOptionalBoolean('rejectUnauthorized'),
+            },
+          }
+        : {}),
+    };
+  }
+  if (config.getOptionalString('provider') === 'aws') {
+    const awsCredentials = await awsGetCredentials();
+    const AWSConnection = createAWSConnection(awsCredentials);
+    return {
+      provider: 'aws',
+      node: config.getString('node'),
+      ...AWSConnection,
+      ...(sslConfig
+        ? {
+            ssl: {
+              rejectUnauthorized:
+                sslConfig?.getOptionalBoolean('rejectUnauthorized'),
+            },
+          }
+        : {}),
+    };
+  }
+  const authConfig = config.getOptionalConfig('auth');
+  const auth =
+    authConfig &&
+    (authConfig.has('apiKey')
+      ? {
+          apiKey: authConfig.getString('apiKey'),
+        }
+      : {
+          username: authConfig.getString('username'),
+          password: authConfig.getString('password'),
+        });
+  return {
+    provider: 'generic',
```

Comment:
I'd recommend sticking with the types/options set in `config.d.ts` and just not returning a provider, here (rather than adding this `generic`)